web/client/codechecker_client/cmd/store.py

```diff
@@ -559,6 +578,24 @@ def parse_report_files(report_files):
                 report_file_info.main_report_positions)
             missing_source_files = \
                 missing_source_files | source_in_reports.missing
+    else:
+        with concurrent.futures.ProcessPoolExecutor() as executor:
+            for report_f, v in zip(report_files,
+                                executor.map(parse_collect_plist_info,
+                                                report_files)):
+
+                report_file_info, source_in_reports = v
+
+                if report_file_info.store_it:
+                    files_to_compress.add(report_f)
+
+                source_file_info.update(source_in_reports.source_info)
+                changed_files = \
+                    changed_files | source_in_reports.changed_since_report_gen
+                main_report_positions.extend(
+                    report_file_info.main_report_positions)
+                missing_source_files = \
+                    missing_source_files | source_in_reports.missing
```

Comment:
Is usage of `nonlocal` allowed here ?

Reply:
i did submit a fix for this but, i'd assume things would be cleaner if i could use nonlocal in nested function to avoid passing and returning the variables back and forth to loop body implementation..